### PR TITLE
Use assignment creator's token to create repo in certain cases

### PR DESCRIPTION
**Pull request by EricPickup**
## What
Back-end solution to #2222 

If the starter code repository is private and does not belong to the classroom organization, use the assignment creator's token to create the student repo.

In all other cases, continue our current method of grabbing a random teachers token to create the students repositories.

### CODE REVIEW

File: spec/services/create_github_repo_service_spec.rb [hunk 1]
```diff
@@ -48,8 +48,7 @@
         options = {
           private: false,
           is_template: true,
-          auto_init: true,
-          accept: "application/vnd.github.baptiste-preview"
+          auto_init: true
```
Comment (EricPickup) [inline spec/services/create_github_repo_service_spec.rb]: These changes are unrelated to the PR, just removing un-needed header!

File: app/services/create_github_repo_service.rb [hunk 2]
```diff
@@ -248,6 +248,16 @@ def report_template_error_to_failbot(error, options)
...
 
+  def github_organization_with_access
+    github_client = assignment.creator.github_client
+    starter_code_repository = github_client.repository(assignment.starter_code_repo_id)
+
+    return organization.github_organization if !starter_code_repository.private ||
+        starter_code_repository.owner.login == exercise.organization_login
+
+    GitHubOrganization.new(github_client, starter_code_repository)
+  end
```
Comment (shaunakpp) [inline app/services/create_github_repo_service.rb]: Awesome! 

Can we use a `attr_accessor`  like `@organization_client` for this? Use the logic to initialize the correct `GitHubOrganization` object, and use that instead of a method call? It can also help us out in testing where we can mock/stub the object.

Another idea would be:

We currently delegate `organization` to `Exercise`, so maybe we can move the logic there? Something on the lines of `Exercise#find_github_organization_with_access(starter_code_repository)` which will modify the `@github_organization` `attr_accessor` on `Exercise` and then we can use it in this service everywhere.

Comment (EricPickup) [reply app/services/create_github_repo_service.rb]: Good idea! I combined both your ideas into an instance variable `@github_organization` on `Exercise`

**Comment by EricPickup** [comment]
@shaunakpp Sorry for the force-push, but let me know how this looks!

**Comment by EricPickup** [comment]
Closing this and moving to #2286 where there are no unrelated changes/cassettes
